lib/data/shared/service/remote/interceptors/unconfirmed_interceptor.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:dio/dio.dart';
+import 'package:flutter_template/data/auth/model/auth_data_error.dart';
+import 'package:flutter_template/data/auth/service/remote/model/auth_api_error.dart';
+import 'package:logging_flutter/logging_flutter.dart';
+
+class UnconfirmedInterceptor extends Interceptor {
+  final Future<void> Function() _onUnconfirmed;
```

Comment:
Not sure if this should be an app-wide interceptor.

The `onUnauthorized` pattern already felt a bit like a hack, it's not something we should continue using, as the logic ends up in the `Dependencies` class whose responsibility should solely be to handle DI.

This can only happen on the Login flow right? So it may be best handled on the Bloc there. Maybe as a `mixin`/`inheritance` if it's available in other places (2 or 3).